### PR TITLE
feat: run command

### PR DESCRIPTION
**Pull request by vansangpfiev**
https://github.com/janhq/cortex/issues/1076

```
.\cortex-cpp.exe run tinyllama:gguf
20240829 04:37:36.360000 UTC 15764 INFO  Downloading item: tinyllama-gguf - download_service.cc:35

20240829 04:38:08.108000 UTC 15764 INFO  Downloading item: cortex.llamacpp - download_service.cc:35

Inorder to exit, type `exit()`
> How are you today?
I'm fine. How about you?
> exit()
```

### CODE REVIEW

File: engine/commands/engine_init_cmd.cc [hunk 1]
```diff
@@ -103,36 +103,45 @@ void EngineInitCmd::Exec() const {
...
+              std::filesystem::path downloadedEnginePath{absolute_path};
+              LOG_INFO << "Downloaded engine path: "
+                       << downloadedEnginePath.string();
+
+              archive_utils::ExtractArchive(
+                  downloadedEnginePath.string(),
+                  downloadedEnginePath.parent_path().parent_path().string());
+
+              // remove the downloaded file
+              // TODO(any) Could not delete file on Windows because it is currently hold by httplib(?)
```
Comment (namchuai) [inline engine/commands/engine_init_cmd.cc:119]: macos is able to remove the file though.

File: engine/utils/model_callback_utils.h [hunk 2]
```diff
@@ -40,10 +40,11 @@ inline void DownloadModelCb(const std::string& path) {
     std::string yml_path(path_obj.parent_path().parent_path().string() + "/" +
                          model_config.id + ".yaml");
     std::string yaml_path(path_obj.parent_path().string() + "/model.yml");
-    if (!std::filesystem::exists(yml_path)) { // if model.yml doesn't exsited
+    if (!std::filesystem::exists(yml_path)) {  // if model.yml doesn't exsited
```
Comment (namchuai) [inline engine/utils/model_callback_utils.h]: typo: exist :D

File: engine/utils/model_callback_utils.h [hunk 3]
```diff
@@ -40,10 +40,11 @@ inline void DownloadModelCb(const std::string& path) {
...
       yaml_handler.WriteYamlFile(yml_path);
     }
-    if (!std::filesystem::exists(yaml_path)) {// if <model_id>.yaml doesn't exsited
+    if (!std::filesystem::exists(
+            yaml_path)) {  // if <model_id>.yaml doesn't exsited
```
Comment (namchuai) [inline engine/utils/model_callback_utils.h]: typo: exist :D

File: engine/utils/cortexso_parser.h [hunk 4]
```diff
@@ -44,8 +44,8 @@ inline std::optional<DownloadTask> getDownloadTask(
           downloadItems.push_back(downloadItem);
         }
 
-        DownloadTask downloadTask{};
-        downloadTask.id = modelId;
+        DownloadTask downloadTask{};
```
Comment (namchuai) [inline engine/utils/cortexso_parser.h]: maybe format code issue?

**Review by namchuai — APPROVED** [review]
looks good to me. Just some nit picking. 💯

**Review by nguyenhoangthuan99 — APPROVED** [review]

**Pull request merged**